### PR TITLE
Add experimental LSP in order to support Declarative Gradle's DCL files

### PR DESCRIPTION
**Pull request by hegyibalint**
### Context

Declarative Gradle introduces a new file format `dcl`.
As this file is based on our DCL language, we need extra tooling to support it.

Currently, we are developing an [LSP](https://github.com/gradle/declarative-lsp) to handle the file, and this PR introduces a new plugin using this server.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,62 @@
...
+> This installation method is preferred if you don't want to develop the plugins.
+
+1. Open the "Help > Install New Software..." dialog
+2. In the "Work with:" field, paste the following link: https://hegyibalint.github.io/buildship/
+3. Under "Buildship Extras - Incubating" there should be a new option: _"Buildship - Gradle Declarative editor support"_
+4. Select the option and proceed with the installation
+5. After installation, go to the "Settings > Gradle > Experimental features" menu, and select the `lsp-all.jar` file (the file is the result of `shadowJar` assembled by [declarative-lsp](https://github.com/gradle/declarative-lsp), see that repository for further instructions)
+
+The installation can be kept up-to-date, just go to "Help > Check for Updates".
+If there is a new snapshot version of the plugin, the dialog will offer the new version.
```
Comment (hegyibalint) [inline README.md:21]: :x: This needs to go, I will delete it

File: org.eclipse.buildship.dcl.feature/category.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<site>
+   <bundle id="org.eclipse.buildship.dcl.provider">
+      <category name="Kotlin DSL support"/>
```
Comment (hegyibalint) [inline org.eclipse.buildship.dcl.feature/category.xml:4]: :x: I've made a mistake here
```suggestion
      <category name="Declarative DCL support"/>
```

File: README.md (hunk 1 above)
Comment (donat) [reply README.md:21]: Just add a line here and reference a dedicated document in the `docs` directory.

Comment (hegyibalint) [reply README.md]: Moved to `docs/`

Comment (hegyibalint) [reply README.md]: Pushed a single-line TIP admonition about where to find the docs

**Review by donat — APPROVED** [review]

**Review by donat — CHANGES_REQUESTED** [review]
Look good!
Please make sure that you use the user/email that you specified when you signed the Eclipse Contributor Agreement. See the GH validation failure for details.

image

File: docs/user/DeclarativeGradle.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+1. In the Help > Install New Software dialog, add the following update site: https://download.eclipse.org/buildship/updates/latest-snapshot/
+1. Install the "Buildship - Gradle Declarative editor support" under the "Buildship Extras - Incubating" category.
+1. Check out and build the [declarative-lsp](https://github.com/gradle/declarative-lsp) project by running `shadowJar`. The task should make a `lsp-all.jar` file under `build/libs`.
+1. Go to the "Settings > Gradle > Experimental features" menu, and select this `lsp-all.jar` file.
+1. Upon opening any `gradle.dcl` file, the LSP should turn on. Observe the console for any diagnostic messages.
+
+## Sample project
+
+Our main flagship project showcasing the capabilities is the [unified-prototype](https://github.com/gradle/declarative-gradle/tree/main/unified-prototype).
+This project is importable by Buildship (even without the declarative support). The project needs the latest
```
Comment (eskatos) [inline docs/user/DeclarativeGradle.md:21]: The unified-prototype folder in there is not the flagship project showcasing declarative. It contains some supporting plugins and builds that are ~ integration tests for it.

Please point to https://declarative.gradle.org/docs/getting-started/samples/ instead

Comment (hegyibalint) [reply docs/user/DeclarativeGradle.md]: PTAL

**Review by eskatos — APPROVED** [review]

**Comment by hegyibalint** [comment]
I've rebased and reauthored my commit, but still something doesn't add up

**Review by donat — APPROVED** [review]

**Comment by donat** [comment]
@hegyibalint Thank you for the contribution!

**Pull request merged**